lte/gateway/c/oai/lib/s6a_proxy/S6aClient.cpp

```diff
@@ -88,17 +90,64 @@ static bool read_mme_cloud_subscriberdb_enabled(void) {
   return mconfig.cloud_subscriberdb_enabled();
 }
 
-S6aClient& S6aClient::get_instance() {
-  static S6aClient client_instance;
-  return client_instance;
+S6aClient& S6aClient::get_s6a_proxy_instance(bool enable_s6a_proxy_channel) {
+  static S6aClient s6a_proxy_instance(enable_s6a_proxy_channel);
+  return s6a_proxy_instance;
 }
 
-S6aClient::S6aClient() {
-  // Create channel based on relay_enabled and cloud_subscriberdb_enabled
-  // flags. If relay_enabled is true, then create a channel towards the FeG.
+S6aClient& S6aClient::get_subdb_instance(bool enable_s6a_proxy_channel) {
+  static S6aClient subdb_instance(enable_s6a_proxy_channel);
+  return subdb_instance;
+}
+
+// Extract MCC and MNC from the imsi received and match with
+// configuration
+bool match_fed_mode_map(const char* imsi) {
```

Comment:
wouldn't be easier to convert the imsi[0]:imsi[5] to a string and then compare that string to the string on the configuration? That conversion should cause much overhead to cpp

Reply:
AGW uses MCC and MNC as decimal digits which is the 3gpp specified format. After parsing configuration,  MCC and MNC are converted into decimal digits format. Hence this conversion is needed while extracting MCC and MNC from IMSI

Reply:
Understood. 
I know this may not be possible since there is a lot of code in there, but in the future we may want to strip the 3gpp format and convert those values into ints or string so we can manage them properly in the code.